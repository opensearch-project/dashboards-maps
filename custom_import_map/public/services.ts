/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { CoreStart } from '../../../src/core/public';

export const postGeojson = async (requestBody: any, http: CoreStart['http']) => {
  try {
    const response = await http.post('../api/custom_import_map/_upload', {
      body: requestBody,
    });
    return response;
  } catch (e) {
    return e;
  }
};

export const getIndex = async (indexName: string, http: CoreStart['http']) => {
  try {
    const response = await http.post('../api/custom_import_map/_indices', {
      body: JSON.stringify({
        index: indexName,
      }),
    });
    return response;
  } catch (e) {
    return e;
  }
};
