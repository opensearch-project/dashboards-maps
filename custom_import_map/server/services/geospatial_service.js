/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

export default class GeospatialService {
  constructor(driver) {
    this.driver = driver;
  }

  uploadGeojson = async (context, req, res) => {
    try {
      const { callAsCurrentUser } = await this.driver.asScoped(req);
      const uploadResponse = await callAsCurrentUser('geospatial.geospatialQuery', {
        body: req.body,
      });
      return res.ok({
        body: {
          ok: true,
          resp: uploadResponse,
        },
      });
    } catch (err) {
      return res.ok({
        body: {
          ok: false,
          resp: err.message,
        },
      });
    }
  };
}
