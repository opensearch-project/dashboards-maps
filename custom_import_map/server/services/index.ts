/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import GeospatialService from './geospatial_service';
import OpensearchService from './opensearch_service';

export { GeospatialService, OpensearchService };
