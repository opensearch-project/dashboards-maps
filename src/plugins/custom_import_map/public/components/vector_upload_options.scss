/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */
 
.formattedSpan {
    display: block;
}

.importFileButton {
    text-align: right;
    margin: 0 auto;
}
