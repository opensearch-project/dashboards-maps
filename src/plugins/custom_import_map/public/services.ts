/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

export const postGeojson = async (requestBody: any, http: any) => {
  try {
    const response = await http.post('../api/custom_import_map/_upload', {
      body: requestBody,
    });
    return response;
  } catch (e) {
    return e;
  }
};

export const getIndex = async (indexName: string, http: any) => {
  try {
    const response = await http.post('../api/custom_import_map/_indices', {
      body: JSON.stringify({
        index: indexName,
      }),
    });
    return response;
  } catch (e) {
    return e;
  }
};

export const getPlugins = async (http: any) => {
  try {
    const response = await http.post('../api/custom_import_map/_plugins', {});
    return response;
  } catch (e) {
    return e;
  }
};
