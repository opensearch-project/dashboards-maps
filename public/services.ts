/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { CoreStart } from '../../../src/core/public';
import { createGetterSetter } from '../../../src/plugins/opensearch_dashboards_utils/common';
import { TimefilterContract } from '../../../src/plugins/data/public';

export const postGeojson = async (
  requestBody: any,
  http: CoreStart['http'],
  dataSourceRefId: string
) => {
  try {
    const query = dataSourceRefId ? { dataSourceId: dataSourceRefId } : undefined;

    return await http.post('../api/custom_import_map/_upload', {
      body: requestBody,
      ...(query && { query }),
    });
  } catch (e) {
    return e;
  }
};

export const getIndex = async (
  indexName: string,
  http: CoreStart['http'],
  dataSourceRefId: string
) => {
  try {
    const query = dataSourceRefId ? { dataSourceId: dataSourceRefId } : undefined;
    return await http.post('../api/custom_import_map/_indices', {
      body: JSON.stringify({ index: indexName }),
      ...(query && { query }),
    });
  } catch (e) {
    return e;
  }
};

export const [getTimeFilter, setTimeFilter] = createGetterSetter<TimefilterContract>('TimeFilter');
