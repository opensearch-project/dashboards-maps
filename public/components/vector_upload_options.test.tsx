/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import * as React from 'react';
import { VectorUploadOptions } from './vector_upload_options';
import { screen, render, waitFor } from '@testing-library/react';
import { fireEvent } from '@testing-library/dom';
import '@testing-library/jest-dom';
import * as serviceApiCalls from '../services';

jest.mock('../../../../src/plugins/opensearch_dashboards_react/public', () => ({
  useOpenSearchDashboards: jest.fn().mockReturnValue({
    services: {
      http: {
        post: () => {
          Promise.resolve({});
        },
      },
      notifications: {
        toasts: { addSuccess: jest.fn(), addDanger: jest.fn(), addWarning: jest.fn() },
      },
    },
  }),
  toMountPoint: jest.fn().mockReturnValue({}),
}));

describe('vector_upload_options', () => {
  const props = {
    vis: {
      data: {
        indexPattern: {
          dataSourceRef: {
            id: 'mock-data-source-id',
          },
        },
      },
    },
  };

  const getIndexResponseWhenIndexIsNotPresent = {
    ok: false,
    resp: [],
  };

  const getIndexResponseWhenIndexIsPresent = {
    ok: true,
    resp: [
      {
        health: 'yellow',
        index: 'sample-map',
        status: 'open',
      },
    ],
  };

  const failedPostGeojsonResponse = {
    ok: false,
    resp: {},
  };

  const singlePartialFailurePostGeojsonResponse = {
    ok: true,
    resp: {
      took: 1969,
      errors: true,
      total: 3220,
      success: 3219,
      failure: 1,
      failures: [],
    },
  };

  const multiplePartialFailuresPostGeojsonResponse = {
    ok: true,
    resp: {
      took: 1969,
      errors: true,
      total: 3220,
      success: 3218,
      failure: 2,
      failures: [],
    },
  };

  const noDocumentsIndexedPostGeojsonResponse = {
    ok: true,
    resp: {
      took: 1969,
      errors: true,
      total: 3220,
      success: 0,
      failure: 3220,
      failures: [],
    },
  };

  const multipleSuccessfulPostGeojsonResponse = {
    ok: true,
    resp: {
      took: 1969,
      errors: true,
      total: 3220,
      success: 3220,
      failure: 0,
      failures: [],
    },
  };

  const singleSuccessfulPostGeojsonResponse = {
    ok: true,
    resp: {
      took: 1969,
      errors: true,
      total: 1,
      success: 1,
      failure: 0,
      failures: [],
    },
  };

  const vectorUploadOptionsWithIndexNameUtil = (userEnteredIndexName: string, message: string) => {
    render(<VectorUploadOptions {...props} />);
    const indexName = screen.getByTestId('customIndex');
    fireEvent.change(indexName, { target: { value: userEnteredIndexName } });
    const button = screen.getByRole('button', { name: 'import-file-button' });
    fireEvent.click(button);
    expect(screen.getByText(message)).toBeInTheDocument();
  };

  const vectorUploadOptionsWithIndexNameRendererUtil = async (
    userEnteredIndexName: string,
    message: string
  ) => {
    const tree = render(<VectorUploadOptions {...props} />);
    const indexName = screen.getByTestId('customIndex');
    fireEvent.change(indexName, { target: { value: userEnteredIndexName } });
    const button = screen.getByRole('button', { name: 'import-file-button' });
    fireEvent.click(button);
    console.log(tree.getByText('errorIndexName'));
    await expect(tree.findAllByText(message)).toBeTruthy();
  };

  const addUserInputToDOM = async () => {
    const jsonData = {
      type: 'FeatureCollection',
      name: 'sample',
      features: [
        {
          type: 'Feature',
          properties: { name: 'sample' },
          geometry: { type: 'Polygon', coordinates: [] },
        },
      ],
    };
    const { getByTestId } = render(<VectorUploadOptions {...props} />);
    const indexName = screen.getByTestId('customIndex');
    fireEvent.change(indexName, { target: { value: 'sample' } });
    const uploader = getByTestId('filePicker');

    const str = JSON.stringify([jsonData]);
    const blob = new Blob([str]);
    const file = new File([blob], 'sample.json', { type: 'application/JSON' });
    File.prototype.text = jest.fn().mockResolvedValueOnce(JSON.stringify([jsonData]));
    await fireEvent.change(uploader, {
      target: { files: [file] },
    });
    await expect(uploader.files[0].name).toBe('sample.json');
  };

  it('renders the VectorUploadOptions based on props provided', () => {
    const vectorUploadOptions = render(<VectorUploadOptions {...props} />);
    expect(vectorUploadOptions).toMatchSnapshot();
  });

  it('renders the VectorUploadOptions component with error message when index name is invalid', () => {
    vectorUploadOptionsWithIndexNameUtil(
      '+abc',
      "Map name can't start with + , _ , - or . It should start with a-z."
    );
  });

  it('renders the VectorUploadOptions component with error message when index name is greater than 250 characters', () => {
    vectorUploadOptionsWithIndexNameUtil(
      'berhtoe7k9yyl43uuzlh6hqsc00iunkqu49110u3kxizck9hy6f584mfaksjcx3zekntyid2tqy39msp25kp0r1gnib5noqmtz1hatq3s4lsbluwrfljrglt7sg3fp1uebukm1ycvh1onrylwrogclvhpf7npzhcfbrvcybmofee5sflwnsx2xxkgqjfsrsg7nz032jlmm0cpahltdekhyg66pcv2plukby8fgm3vze9jhewrilre07kdakb0ul7',
      'Map name should be less than 250 characters.'
    );
  });

  it('renders the VectorUploadOptions component with error message when index name has upper case letters', async () => {
    try {
      await vectorUploadOptionsWithIndexNameRendererUtil(
        'ABC',
        'Upper case letters are not allowed'
      );
    } catch (err) {}
  });

  it('renders the VectorUploadOptions component with error message when index name has special characters', async () => {
    try {
      await vectorUploadOptionsWithIndexNameRendererUtil(
        'a#bc',
        'Special characters are not allowed.'
      );
    } catch (err) {}
  });

  it('renders the VectorUploadOptions component with error message when index name has -map as suffix', async () => {
    try {
      await vectorUploadOptionsWithIndexNameRendererUtil(
        'sample-map',
        "Map name can't end with -map."
      );
    } catch (err) {}
  });

  it('renders the VectorUploadOptions component when we have successfully indexed all the data having multiple features', async () => {
    addUserInputToDOM();
    console.log('test case for successfully indexed file data');
    const button = screen.getByRole('button', { name: 'import-file-button' });
    jest.spyOn(serviceApiCalls, 'getIndex').mockImplementation(() => {
      return Promise.resolve(getIndexResponseWhenIndexIsNotPresent);
    });
    jest.spyOn(serviceApiCalls, 'postGeojson').mockImplementation(() => {
      return Promise.resolve(multipleSuccessfulPostGeojsonResponse);
    });
    await waitFor(() => {
      fireEvent.click(button);
    });
  });

  it('renders the VectorUploadOptions component when we have successfully indexed the data having single feature', async () => {
    addUserInputToDOM();
    console.log('test case for successfully indexed file data');
    const button = screen.getByRole('button', { name: 'import-file-button' });
    jest.spyOn(serviceApiCalls, 'getIndex').mockImplementation(() => {
      return Promise.resolve(getIndexResponseWhenIndexIsNotPresent);
    });
    jest.spyOn(serviceApiCalls, 'postGeojson').mockImplementation(() => {
      return Promise.resolve(singleSuccessfulPostGeojsonResponse);
    });
    await waitFor(() => {
      fireEvent.click(button);
    });
  });

  it('renders the VectorUploadOptions component when we have a single failure during indexing', async () => {
    addUserInputToDOM();
    console.log('test case for partial failures during indexing');
    const button = screen.getByRole('button', { name: 'import-file-button' });
    jest.spyOn(serviceApiCalls, 'getIndex').mockImplementation(() => {
      return Promise.resolve(getIndexResponseWhenIndexIsNotPresent);
    });
    jest.spyOn(serviceApiCalls, 'postGeojson').mockImplementation(() => {
      return Promise.resolve(singlePartialFailurePostGeojsonResponse);
    });
    await waitFor(() => {
      fireEvent.click(button);
    });
  });

  it('renders the VectorUploadOptions component when we have multiple partial failures during indexing', async () => {
    addUserInputToDOM();
    console.log('test case for partial failures during indexing');
    const button = screen.getByRole('button', { name: 'import-file-button' });
    jest.spyOn(serviceApiCalls, 'getIndex').mockImplementation(() => {
      return Promise.resolve(getIndexResponseWhenIndexIsNotPresent);
    });
    jest.spyOn(serviceApiCalls, 'postGeojson').mockImplementation(() => {
      return Promise.resolve(multiplePartialFailuresPostGeojsonResponse);
    });
    await waitFor(() => {
      fireEvent.click(button);
    });
  });

  it('renders the VectorUploadOptions component when all the documents fail to index', async () => {
    addUserInputToDOM();
    console.log('test case for failed documents');
    const button = screen.getByRole('button', { name: 'import-file-button' });
    jest.spyOn(serviceApiCalls, 'getIndex').mockImplementation(() => {
      return Promise.resolve(getIndexResponseWhenIndexIsNotPresent);
    });
    jest.spyOn(serviceApiCalls, 'postGeojson').mockImplementation(() => {
      return Promise.resolve(noDocumentsIndexedPostGeojsonResponse);
    });
    await waitFor(() => {
      fireEvent.click(button);
    });
  });

  it('renders the VectorUploadOptions component when postGeojson call fails', async () => {
    addUserInputToDOM();
    console.log('test case for call failure to postGeojson');
    const button = screen.getByRole('button', { name: 'import-file-button' });
    jest.spyOn(serviceApiCalls, 'getIndex').mockImplementation(() => {
      return Promise.resolve(getIndexResponseWhenIndexIsNotPresent);
    });
    jest.spyOn(serviceApiCalls, 'postGeojson').mockImplementation(() => {
      return Promise.resolve(failedPostGeojsonResponse);
    });
    await waitFor(() => {
      fireEvent.click(button);
    });
  });

  it('renders the VectorUploadOptions component when getIndex returns a duplicate index', async () => {
    addUserInputToDOM();
    console.log('test case for duplicate index check');
    const button = screen.getByRole('button', { name: 'import-file-button' });
    jest.spyOn(serviceApiCalls, 'getIndex').mockImplementation(() => {
      return Promise.resolve(getIndexResponseWhenIndexIsPresent);
    });
    await waitFor(() => {
      fireEvent.click(button);
    });
  });

  it('renders the VectorUploadOptions component when uploaded file size is zero', async () => {
    const { getByTestId } = render(<VectorUploadOptions {...props} />);
    const indexName = screen.getByTestId('customIndex');
    fireEvent.change(indexName, { target: { value: 'sample' } });
    const uploader = getByTestId('filePicker');
    const file = new File([], 'sample.json', { type: 'application/JSON' });
    File.prototype.text = jest.fn().mockResolvedValueOnce(JSON.stringify([]));
    await fireEvent.change(uploader, {
      target: { files: [file] },
    });
    await expect(uploader.files[0].name).toBe('sample.json');
    const button = screen.getByRole('button', { name: 'import-file-button' });
    jest.spyOn(serviceApiCalls, 'getIndex').mockImplementation(() => {
      return Promise.resolve(getIndexResponseWhenIndexIsNotPresent);
    });
    jest.spyOn(serviceApiCalls, 'postGeojson').mockImplementation(() => {
      return Promise.resolve(multipleSuccessfulPostGeojsonResponse);
    });
    await waitFor(() => {
      fireEvent.click(button);
    });
  });

  it('renders danger toast as file format is not geojson or json', async () => {
    const { getByTestId } = render(<VectorUploadOptions {...props} />);
    const indexName = screen.getByTestId('customIndex');
    fireEvent.change(indexName, { target: { value: 'sample' } });
    const uploader = getByTestId('filePicker');
    const file = new File([], 'sample.txt', { type: 'text/plain' });
    File.prototype.text = jest.fn().mockResolvedValueOnce(JSON.stringify([]));
    await fireEvent.change(uploader, {
      target: { files: [file] },
    });
    await expect(uploader.files[0].name).toBe('sample.txt');
  });

  it('renders the VectorUploadOptions component when uploaded file size is >25 MB', async () => {
    const { getByTestId } = render(<VectorUploadOptions {...props} />);
    const indexName = screen.getByTestId('customIndex');
    fireEvent.change(indexName, { target: { value: 'sample' } });
    const uploader = getByTestId('filePicker');
    const jsonData = {
      type: 'FeatureCollection',
      name: 'sample',
      features: [
        {
          type: 'Feature',
          properties: { name: 'sample' },
          geometry: { type: 'Polygon', coordinates: [] },
        },
      ],
    };
    const jsonDataArray = [];
    const max = 100000;
    for (; jsonDataArray.push(jsonData) < max; );
    const str = JSON.stringify(jsonDataArray);
    const blob = new Blob([str]);
    const file = new File([blob], 'sample.json', { type: 'application/JSON' });
    File.prototype.text = jest.fn().mockResolvedValueOnce(JSON.stringify(jsonDataArray));
    await fireEvent.change(uploader, {
      target: { files: [file] },
    });
    await expect(uploader.files[0].name).toBe('sample.json');
  });
});
