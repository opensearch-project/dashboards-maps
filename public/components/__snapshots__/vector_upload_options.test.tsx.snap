// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vector_upload_options renders the VectorUploadOptions based on props provided 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        id="uploadCustomVectorMap"
      >
        <div
          aria-label="import-vector-map-card"
          class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow euiCard euiCard--leftAligned"
        >
          <div
            class="euiCard__content"
          >
            <span
              class="euiTitle euiTitle--small euiCard__title"
              id="generated-idTitle"
            />
            
            <div
              class="euiCard__children"
            >
              <div
                aria-label="medium-spacer"
                class="euiSpacer euiSpacer--s"
              />
              <div
                aria-label="upload-map-text"
                class="euiText euiText--small"
              >
                Upload map
              </div>
              <div
                aria-label="medium-spacer"
                class="euiSpacer euiSpacer--m"
              />
              <div
                aria-label="form-row-for-file-picker"
                class="euiFormRow euiFormRow--compressed"
                id="generated-id-row"
              >
                <div
                  class="euiFormRow__fieldWrapper"
                >
                  <div
                    class="euiFilePicker euiFilePicker--large euiFilePicker--compressed"
                  >
                    <div
                      class="euiFilePicker__wrap"
                    >
                      <input
                        accept=".json,.geojson"
                        aria-describedby="generated-id-filePicker__prompt"
                        aria-label="geojson-file-picker"
                        class="euiFilePicker__input"
                        data-test-subj="filePicker"
                        data-testid="filePicker"
                        id="generated-id"
                        required=""
                        type="file"
                      />
                      <div
                        class="euiFilePicker__prompt"
                        id="generated-id-filePicker__prompt"
                      >
                        <svg
                          aria-hidden="true"
                          class="euiIcon euiIcon--large euiIcon-isLoading euiFilePicker__icon"
                          focusable="false"
                          height="16"
                          role="img"
                          viewBox="0 0 16 16"
                          width="16"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M5.277 10.088c.02.014.04.03.057.047.582.55 1.134.812 1.666.812.586 0 1.84-.293 3.713-.88L9 6.212V2H7v4.212l-1.723 3.876Zm-.438.987L3.539 14h8.922l-1.32-2.969C9.096 11.677 7.733 12 7 12c-.74 0-1.463-.315-2.161-.925ZM6 2H5V1h6v1h-1v4l3.375 7.594A1 1 0 0 1 12.461 15H3.54a1 1 0 0 1-.914-1.406L6 6V2Z"
                          />
                        </svg>
                        <div
                          class="euiFilePicker__promptText"
                        >
                          Select or drag and drop a json file
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                aria-label="medium-spacer"
                class="euiSpacer euiSpacer--m"
              />
              <div
                aria-label="geojson-file-format-text"
                class="euiText euiText--extraSmall"
              >
                <div
                  class="euiTextColor euiTextColor--subdued"
                >
                  <p>
                    <span
                      class="formattedSpan"
                    >
                      Formats accepted: .json, .geojson
                    </span>
                    <span
                      class="formattedSpan"
                    >
                      Max size: 25 MB
                    </span>
                    <span
                      class="formattedSpan"
                    >
                      Coordinates must be in EPSG:4326 coordinate reference system.
                    </span>
                  </p>
                </div>
              </div>
              <div
                aria-label="medium-spacer"
                class="euiSpacer euiSpacer--m"
              />
              <div
                aria-label="map-name-prefix-text"
                class="euiText euiText--small"
              >
                Map name prefix
              </div>
              <div
                aria-label="medium-spacer"
                class="euiSpacer euiSpacer--m"
              />
              <div
                class="euiFormControlLayout euiFormControlLayout--compressed"
              >
                <div
                  class="euiFormControlLayout__childrenWrapper"
                >
                  <input
                    aria-label="map-name-text-field"
                    class="euiFieldText euiFieldText--compressed"
                    data-test-subj="customIndex"
                    data-testid="customIndex"
                    id="customIndex"
                    label="Map name"
                    name="customIndex"
                    placeholder="Enter a valid map name prefix"
                    required=""
                    tabindex="0"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                aria-label="medium-spacer"
                class="euiSpacer euiSpacer--m"
              />
              <div
                aria-label="map-name-guidelines-text"
                class="euiText euiText--extraSmall"
              >
                <div
                  class="euiTextColor euiTextColor--subdued"
                >
                  Map name guidleines:
                  <ul>
                    <li>
                       Map name prefix must contain 1-250 characters.
                    </li>
                    <li>
                       Map name prefix must start with a-z.
                    </li>
                    <li>
                       Valid characters are a-z, 0-9, - and _ .
                    </li>
                    <li>
                      The final map name will be the entered prefix here followed by -map as the suffix.
                    </li>
                  </ul>
                </div>
              </div>
              <div
                aria-label="map-name-error-text"
                class="euiText euiText--extraSmall"
              >
                <span
                  aria-label="map-name-error-text-color"
                  class="euiTextColor euiTextColor--danger"
                >
                  <p
                    name="errorIndexName"
                  />
                </span>
              </div>
              <div
                aria-label="medium-spacer"
                class="euiSpacer euiSpacer--m"
              />
              <div
                class="importFileButton"
              >
                <button
                  aria-label="import-file-button"
                  class="euiButton euiButton--primary euiButton--small"
                  id="submitButton"
                  type="button"
                >
                  <span
                    class="euiButtonContent euiButton__content"
                  >
                    <span
                      class="euiButton__text"
                    >
                      Import file
                    </span>
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      id="uploadCustomVectorMap"
    >
      <div
        aria-label="import-vector-map-card"
        class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow euiCard euiCard--leftAligned"
      >
        <div
          class="euiCard__content"
        >
          <span
            class="euiTitle euiTitle--small euiCard__title"
            id="generated-idTitle"
          />
          
          <div
            class="euiCard__children"
          >
            <div
              aria-label="medium-spacer"
              class="euiSpacer euiSpacer--s"
            />
            <div
              aria-label="upload-map-text"
              class="euiText euiText--small"
            >
              Upload map
            </div>
            <div
              aria-label="medium-spacer"
              class="euiSpacer euiSpacer--m"
            />
            <div
              aria-label="form-row-for-file-picker"
              class="euiFormRow euiFormRow--compressed"
              id="generated-id-row"
            >
              <div
                class="euiFormRow__fieldWrapper"
              >
                <div
                  class="euiFilePicker euiFilePicker--large euiFilePicker--compressed"
                >
                  <div
                    class="euiFilePicker__wrap"
                  >
                    <input
                      accept=".json,.geojson"
                      aria-describedby="generated-id-filePicker__prompt"
                      aria-label="geojson-file-picker"
                      class="euiFilePicker__input"
                      data-test-subj="filePicker"
                      data-testid="filePicker"
                      id="generated-id"
                      required=""
                      type="file"
                    />
                    <div
                      class="euiFilePicker__prompt"
                      id="generated-id-filePicker__prompt"
                    >
                      <svg
                        aria-hidden="true"
                        class="euiIcon euiIcon--large euiIcon-isLoading euiFilePicker__icon"
                        focusable="false"
                        height="16"
                        role="img"
                        viewBox="0 0 16 16"
                        width="16"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M5.277 10.088c.02.014.04.03.057.047.582.55 1.134.812 1.666.812.586 0 1.84-.293 3.713-.88L9 6.212V2H7v4.212l-1.723 3.876Zm-.438.987L3.539 14h8.922l-1.32-2.969C9.096 11.677 7.733 12 7 12c-.74 0-1.463-.315-2.161-.925ZM6 2H5V1h6v1h-1v4l3.375 7.594A1 1 0 0 1 12.461 15H3.54a1 1 0 0 1-.914-1.406L6 6V2Z"
                        />
                      </svg>
                      <div
                        class="euiFilePicker__promptText"
                      >
                        Select or drag and drop a json file
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-label="medium-spacer"
              class="euiSpacer euiSpacer--m"
            />
            <div
              aria-label="geojson-file-format-text"
              class="euiText euiText--extraSmall"
            >
              <div
                class="euiTextColor euiTextColor--subdued"
              >
                <p>
                  <span
                    class="formattedSpan"
                  >
                    Formats accepted: .json, .geojson
                  </span>
                  <span
                    class="formattedSpan"
                  >
                    Max size: 25 MB
                  </span>
                  <span
                    class="formattedSpan"
                  >
                    Coordinates must be in EPSG:4326 coordinate reference system.
                  </span>
                </p>
              </div>
            </div>
            <div
              aria-label="medium-spacer"
              class="euiSpacer euiSpacer--m"
            />
            <div
              aria-label="map-name-prefix-text"
              class="euiText euiText--small"
            >
              Map name prefix
            </div>
            <div
              aria-label="medium-spacer"
              class="euiSpacer euiSpacer--m"
            />
            <div
              class="euiFormControlLayout euiFormControlLayout--compressed"
            >
              <div
                class="euiFormControlLayout__childrenWrapper"
              >
                <input
                  aria-label="map-name-text-field"
                  class="euiFieldText euiFieldText--compressed"
                  data-test-subj="customIndex"
                  data-testid="customIndex"
                  id="customIndex"
                  label="Map name"
                  name="customIndex"
                  placeholder="Enter a valid map name prefix"
                  required=""
                  tabindex="0"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              aria-label="medium-spacer"
              class="euiSpacer euiSpacer--m"
            />
            <div
              aria-label="map-name-guidelines-text"
              class="euiText euiText--extraSmall"
            >
              <div
                class="euiTextColor euiTextColor--subdued"
              >
                Map name guidleines:
                <ul>
                  <li>
                     Map name prefix must contain 1-250 characters.
                  </li>
                  <li>
                     Map name prefix must start with a-z.
                  </li>
                  <li>
                     Valid characters are a-z, 0-9, - and _ .
                  </li>
                  <li>
                    The final map name will be the entered prefix here followed by -map as the suffix.
                  </li>
                </ul>
              </div>
            </div>
            <div
              aria-label="map-name-error-text"
              class="euiText euiText--extraSmall"
            >
              <span
                aria-label="map-name-error-text-color"
                class="euiTextColor euiTextColor--danger"
              >
                <p
                  name="errorIndexName"
                />
              </span>
            </div>
            <div
              aria-label="medium-spacer"
              class="euiSpacer euiSpacer--m"
            />
            <div
              class="importFileButton"
            >
              <button
                aria-label="import-file-button"
                class="euiButton euiButton--primary euiButton--small"
                id="submitButton"
                type="button"
              >
                <span
                  class="euiButtonContent euiButton__content"
                >
                  <span
                    class="euiButton__text"
                  >
                    Import file
                  </span>
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
